src/storage/engines/types.ts

```diff
@@ -691,6 +722,40 @@ export abstract class StorageEngine {
     await this.__throttleSaveAnswers(answers);
   }
 
+  // Updates the progress data in the sequence assignment
+  async updateProgressData(
+    progressData: { total: number; answered: string[]; isDynamic: boolean },
+    participantId?: string,
+  ) {
+    if (!this.studyId) {
+      throw new Error('Study ID is not set');
+    }
```

Comment:
This only works for firebase, let's add a check because this function will fail for others
```suggestion
  async updateProgressData(
    progressData: { total: number; answered: string[]; isDynamic: boolean },
    participantId?: string,
  ) {
    if (!this.studyId) {
      throw new Error('Study ID is not set');
    }
    
    if (this.getEngine() !== 'firebase') {
      return;  
    }
```

Reply:
updated